lib/src/ruby/ruby_value.cc

```diff
@@ -87,7 +87,10 @@ namespace facter { namespace ruby {
             return;
         }
         if (ruby.is_fixednum(value)) {
-            json.SetInt64(ruby.rb_num2long(value));
+            // On Windows 64-bit with Ruby 2.2, rb_num2long returns the wrong value for negative numbers.
+            // FixedNums can only represent a word (32-bit) so explicitly cast to an int.
+            // TODO: Handle BigNum.
+            json.SetInt64(static_cast<int>(ruby.rb_num2long(value)));
```

Comment:
Yes, we just skip calling it for BigNums. That'll be the fix for another FACT ticket.